platforms/documentation/docs/src/samples/writing-tasks/custom-test-task/common/plugin/src/main/java/com/example/CustomTestPlugin.java

```diff
@@ -0,0 +1,15 @@
+package com.example;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+
+class CustomTestPlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        project.getTasks().register("test", CustomTest.class, task -> {
+            task.setGroup("verification");
+            task.setDescription("Runs the tests.");
+            task.getFail().convention(false);
```

Comment:
💭 Having to set a property to make it fail is distracting - 2 separate tasks with 2 separate task types would be clearer.

Reply:
I think it's OK.  I also thought about having two different tasks, but they would be almost identical.